content/extending-chat-widget/custom-identity-provider/index.mdx

```diff
@@ -0,0 +1,183 @@
+---
+weight: 60
+category: "extending-chat-widget"
+subcategory: "custom-identity-provider"
+title: "Custom Identity Provider"
+tagline: "Pair your own customer identity with LiveChat"
+desc: "Cross-platform chats for your applications"
+versionGroup: "chat-widget"
+---
+
+# Introduction
+
+## What is it for?
+
+_Customer identity_ is an id thats assigned to each customer by our API. Basing on that we can match data and chat history with certain person using our services.
+
+**Customer Identity Provider** was created for those who build more advanced integrations that handle user sessions. It allows you to match customer identities in LiveChat with those stored in your service. As a result, your customers will never lose chats they had with your support. What is more, if you also use LiveChat on other platforms (for example, in mobile applications), customer conversations will sync, making the communication even more seamless and natural. Mind the fact that _customer identity_ is not customer metadata such as name or email. If you would like to pass one of those, you are probably looking for [updating session variables](/extending-chat-widget/javascript-api/#update-session-variables).
+
+## Setup
+Custom Identity Provider is essentially a singleton function. You can define it as a part of the `LiveChatWidget` object from the tracking code used to install ChatWidget on your website or any other place accessing `window.__lc` object. However, **it has to be provided before widget initialization**.
```

Comment:
```suggestion

Custom Identity Provider is essentially a singleton function. You can define it as a part of the `LiveChatWidget` object from the tracking code used to install the Chat Widget on your website or any other place accessing `window.__lc` object. However, **it has to be provided before widget initialization**.
```

Reply:
I' not sure about "...or any other place accessing `window.__lc` object"
Do you mean that one can access window.__lc anywhere on the page that has the tracking code? 

My problem is that "any other place" is quite vague.

Reply:
> Do you mean that one can access window.__lc anywhere on the page that has the tracking code?
yes

Reply:
Its quite important to emphasise imo. Especially if someone uses more advanced setup than plain html page (like react or vue app etc.)

Reply:
I also think we should keep it; I'm just thinking about a good way to rephrase it.

Reply:
I'm thinking about a bullet list. Let me know what you think.

Custom Identity Provider is essentially a singleton function. You can define it as part of one of the following objects: 
- the `LiveChatWidget` object from the LiveChat tracking code
- the `window.__lc` object you can access anywhere on a page that has the LiveChat tracking code

In both cases, **it has to be provided before widget initialization**.

Reply:
they are both the same thing actually, its called LiveChatWidget object in https://developers.livechat.com/docs/extending-chat-widget/javascript-api/#getting-started but its assigned to window.__lc property. By splitting it Im afraid we will intorduce some difference that doesn't exist. How about making `LiveChatWidget object` a link to the one I provided here?

Reply:
No splitting, then :)

> You can define it as part of the `LiveChatWidget` object from the LiveChat tracking code used on your website or from any other place that has access to the `window._lc` object (might be the case in more advanced applications).